cmd/maya-exporter/app/collector/pool/metrics.go

```diff
@@ -58,101 +59,140 @@ func (s *statsFloat64) List() []float64 {
-				Name:      "zpool_list_incomplete_stdout_error",
-				Help:      "Total no of incomplete stdout of zpool list command errors",
-			},
-		),
-	}
+	s.usedCapacityPercent = parseFloat64(stats.UsedCapacityPercent, p.metrics)
+}
+
+func (m *metrics) withSize() *metrics {
+	m.size = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "openebs",
+			Name:      "pool_size",
+			Help:      "Size of pool",
+		},
+	)
+	return m
+}
+
+func (m *metrics) withStatus() *metrics {
+	m.status = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "openebs",
+			Name:      "pool_status",
+			Help:      `Status of pool (0, 1, 2, 3, 4, 5, 6)= {"Offline", "Online", "Degraded", "Faulted", "Removed", "Unavail", "NoPoolsAvailable"}`,
+		},
+		[]string{"pool"},
+	)
+	return m
+}
+func (m *metrics) withUsedCapacity() *metrics {
+	m.usedCapacity = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "openebs",
+			Name:      "used_pool_capacity",
+			Help:      "Capacity used by pool",
+		},
+	)
+	return m
+}
+
+func (m *metrics) withFreeCapacity() *metrics {
+	m.freeCapacity = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "openebs",
+			Name:      "free_pool_capacity",
+			Help:      "Free capacity in pool",
+		},
+	)
+	return m
+}
+
+func (m *metrics) withUsedCapacityPercent() *metrics {
+	m.usedCapacityPercent = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "openebs",
+			Name:      "used_pool_capacity_percent",
+			Help:      "Capacity used by pool in percent",
+		},
+	)
+	return m
+}
+
+func (m *metrics) withParseErrorCounter() *metrics {
+	m.zpoolListParseErrorCounter = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "openebs",
+			Name:      "zpool_list_parse_error_count",
+			Help:      "Total no of parsing errors",
+		},
+	)
+	return m
+}
+
+func (m *metrics) withRejectRequestCounter() *metrics {
+	m.zpoolRejectRequestCounter = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "openebs",
+			Name:      "zpool_list_reject_request_count",
+			Help:      "Total no of rejected requests of zpool command",
+		},
+	)
+	return m
+}
+
+func (m *metrics) withCommandErrorCounter() *metrics {
+	m.zpoolCommandErrorCounter = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "openebs",
+			Name:      "zpool_list_command_error",
+			Help:      "Total no of zpool command errors",
+		},
+	)
+	return m
+}
+
+func (m *metrics) withNoPoolAvailableErrorCounter() *metrics {
+	m.noPoolAvailableErrorCounter = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "openebs",
+			Name:      "zpool_list_no_pool_available_error",
+			Help:      "Total no of no pool available errors",
+		},
+	)
+	return m
+}
+
+func (m *metrics) withIncompleteOutputErrorCounter() *metrics {
+	m.inCompleteOutputErrorCounter = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "openebs",
+			Name:      "zpool_list_incomplete_stdout_error",
+			Help:      "Total no of incomplete stdout errors",
```

Comment:
What is the relevance of this metric?

Reply:
This is used for debugging purpose since there are no logs